src/main/java/bio/terra/service/snapshot/flight/LockSnapshotStep.java

```diff
@@ -0,0 +1,51 @@
+package bio.terra.service.snapshot.flight;
+
+import bio.terra.service.snapshot.SnapshotDao;
+import bio.terra.service.snapshot.exception.SnapshotLockException;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class LockSnapshotStep implements Step {
+
+    private SnapshotDao snapshotDao;
+    private String snapshotName;
+
+    private static Logger logger = LoggerFactory.getLogger(LockSnapshotStep.class);
+
+    public LockSnapshotStep(SnapshotDao snapshotDao, String snapshotName) {
+        this.snapshotDao = snapshotDao;
+        this.snapshotName = snapshotName;
+    }
+
+    @Override
+    public StepResult doStep(FlightContext context) {
+        try {
+            FlightMap workingMap = context.getWorkingMap();
+            workingMap.put(SnapshotWorkingMapKeys.SNAPSHOT_NAME, snapshotName);
```

Comment:
Same comment here with snapshot name. We know the name at flight construction time, so I don't think we need to put it in the working map.

Reply:
Done, thanks for pointing it out.